MTM-66596 Re-work "Monitoring the MQTT Service" M&Ms user documentation according to the current MQTT Service implementation (#4564)
SoftwareAG/c8y-docs: content/standard-tenant/monitoring-bundle/messaging-service.md

### Monitoring the MQTT Service {#monitoring-the-mqtt-service}

The MQTT Service creates a fixed set of topics per tenant for routing device traffic.
These topics are the primary way to monitor whether the MQTT Service data flow is working correctly.

{{< c8y-admon-important >}}
These topics are shared across all devices in the tenant. Any action that modifies subscribers affects the message backlog on that topic for all devices in the tenant.
{{< /c8y-admon-important >}}

#### Topics {#mqtt-service-topics}

The MQTT Service creates the following topics:

| Topic                   | Description                                                                                                                                                              |
|-------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `to-device`             | Messages to be sent to devices. Topic contains messages for all the outbound traffic to devices excluding {{< product-c8y-iot >}} SmartREST topics.                      |

Reviewer: ```suggestion
| `to-device`             | Messages to be sent to devices. Topic contains messages for all outbound traffic to devices excluding {{< product-c8y-iot >}} SmartREST topics.                      |
```

> | `from-device`           | Messages arriving from devices to the MQTT Service. Topic contains messages for all the inbound traffic from devices excluding {{< product-c8y-iot >}} SmartREST topics. |

Reviewer: ```suggestion
| `from-device`           | Messages arriving from devices to the MQTT Service. Topic contains messages for all inbound traffic from devices excluding {{< product-c8y-iot >}} SmartREST topics. |
```

| `smartrest-from-device` | Internal topic used to forward SmartREST traffic from devices to the {{< product-c8y-iot >}} core for processing.                                                        |
| `smartrest-to-device`   | Internal topic used to forward SmartREST traffic from the {{< product-c8y-iot >}} core back to devices.                                                                  |

#### Subscribers {#mqtt-service-subscribers}

The expected subscribers for each topic are:

- **`from-device`**: One or more subscribers used in custom microservice implementations or Streaming Analytics. This is the most important topic to monitor for custom integrations.
- **`to-device`**, **`smartrest-to-device`**: One subscriber per MQTT Service instance, for example, `c8y-mqtt-service-0`, `c8y-mqtt-service-1`.

> - **`smartrest-from-device`**: One subscriber per {{< product-c8y-iot >}} core instance, for example, `cumulocity-core`.

Reviewer 2: Suggestion:
```suggestion
- **`smartrest-from-device`**: One subscriber with one consumer per {{< product-c8y-iot >}} core instance, for example, `cumulocity-core`.
```

#### Clear the backlog {#mqtt-service-clear-the-backlog}

When the Messaging Service backlog is full, no new messages can be added to the backlog until it is cleared.
In this situation, client behavior depends on the MQTT protocol version used:
* An MQTT client using protocol version 3.1.1 will simply be disconnected.
* An MQTT client using protocol version 5 will get a negative PUBACK response with `0x97` (quota exceeded) reason code, but it will still remain connected.

Implementations connecting to the MQTT Service must be aware of this and handle these errors appropriately.

{{< c8y-admon-caution >}}
Clearing the backlog by unsubscribing a subscriber removes messages for **all devices** on that topic, not for a specific device.
{{< /c8y-admon-caution >}}

For the `to-device`, `smartrest-from-device`, and `smartrest-to-device` topics, do not attempt to clear the backlog manually.
These subscribers are managed by the MQTT Service and {{< product-c8y-iot >}} core, which re-create them automatically after unsubscription.
If the backlog is full on one of these topics, contact [product support](/additional-resources/contacting-support/).

> ##### Clear the `from-device` backlog {#mqtt-service-clear-from-device-backlog}

Reviewer: ```suggestion
##### Clear the from-device backlog {#mqtt-service-clear-from-device-backlog}
```

Note that the MQTT Service creates a fixed set of topics per tenant (for example, `to-device`, `from-device`, `smartrest-from-device`, `smartrest-to-device`). A high topic count in the MQTT Service section is not expected and may indicate an issue.

#### What should I do when encountering a high number of subscribers?

If you have a single microservice or a single client consuming messages from the Messaging Service, you should typically only have a single subscriber.
Check if the subscriber name used by your client is unique and reused consistently when connecting to the Messaging Service.
A common pitfall is generating a random subscriber name each time a new connection to the Messaging Service is established.

Multiple subscribers are expected when multiple distinct clients consume from the same topic or when using [shared consumer tokens](https://{{< domain-c8y >}}/api/core/#section/Overview/Shared-consumer-tokens).

For the MQTT Service `from-device` topic, the number of subscribers depends on your implementation.
If you have a single custom microservice or Streaming Analytics flow consuming device messages, you should see a single subscriber.

> For the `to-device`, `smartrest-from-device`, and `smartrest-to-device` topics, the number of subscribers should match the number of MQTT Service or {{< product-c8y-iot >}} core instances running in your environment.

Reviewer 2: ```suggestion
For the `to-device` and `smartrest-to-device` topics, the number of subscribers should match the number of MQTT Service instances running in your environment.
For the `smartrest-from-device` topic, there should be one subscriber with a number of consumers that matches the number of {{< product-c8y-iot >}} core instances in your environment.
These consumers will partition the messages on the topic across the core instances so that each message is processed by a single core.
```
